docs/structures/examples.md

```diff
@@ -0,0 +1,132 @@
+This document describes how examples can be authored and will be processed. It's in three sections:
+
+* **Authoring format** describes the formats used by writers to specify examples
+
+* **JSON format** describes the JSON that is built from the content created by the writers
+
+* **Rendering** describes how examples will be rendered as part of an MDN page.
+
+There are three sorts of examples supported:
+
+* **descriptive text only**: these aren't really examples, but enable authors to describe or link to examples if for any reason they don't want to include the example source itself.
+
+* **static code samples**: these consist of blocks of source code, optionally with descriptive text. The source code may not be fully functional (for example, the example might require so much setup or other dependencies that it's not practical to make a functional example).
+
+* **live samples**: these consist of blocks of source code, optionally with descriptive text, and a specification for the size of an iframe in which the result can be presented. The code must be functional: so a consumer can package it into a document and have it run in an iframe.
+
+Note that this doesn't include interactive examples, which are handled separately.
```

Comment:
Why?

Wait, hold on. Let me make sure I understand the distinctions between examples here: 
- descriptive: it's some markdown text?
- static: markdown with some fenced code blocks? Or is this markdown with some extra source files?
- live: these are the older style examples that are in an iframe and have the codepen and jsfiddle buttons?
- interactive: from mdn/interactive-examples?

It's not clear to me what distinction is being made between descriptive and static. It's also not clear to me what makes interactive examples a special case.

Reply:
> Why?
> 
> Wait, hold on. Let me make sure I understand the distinctions between examples here:
> 
> * descriptive: it's some markdown text?
> 
> * static: markdown with some fenced code blocks? Or is this markdown with some extra source files?
> 
> * live: these are the older style examples that are in an iframe and have the codepen and jsfiddle buttons?
> 
> * interactive: from mdn/interactive-examples?
> 

Yes.
 
> It's not clear to me what distinction is being made between descriptive and static.

Well yes this is a bit tenuous. The only point I was trying to make is that examples don't have to contain actual example code. But I could remove it as a separate category and just say that instead.

> It's also not clear to me what makes interactive examples a special case.

Well... interactive examples are specified very differently[1], they live in a different bit of the MDN page, currently there should only be one in the page, and there are quite specific rules for the content they should contain, which is different from other examples.

It might be better to include them in here anyway, but it seemed more natural to list them separately.

[1] this might change: I don't very much like the way the interactive examples editor and sources are completely separate from MDN. We explored using CodePen as the editor for Web API examples, and in that case the examples would start to appear much more like "live samples" from the authoring point of view. But that work has stalled, and I don't know when if ever we will look at it again.